src/main/TreeQuery.cs

```diff
@@ -0,0 +1,202 @@
+﻿using Blazorise;
+using ei8.Cortex.Graph.Common;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web; // Add this for HttpUtility.UrlDecode
+
+namespace ei8.Cortex.Diary.Plugins.Tree
+{
+    public class TreeQuery
+    {
+        [QueryKey(null)]
+        public IEnumerable<Guid> Postsynaptics { get; set; }
+
+        [QueryKey(null)]
+        public Guid RegionId { get; set; }
+
+        [QueryKey("avatarurl")]
+        public string AvatarUrl { get; set; }
+
+        [QueryKey("eupm")]
+        public IEnumerable<Guid> ExpandUntilPostsynapticMirrors { get; set; }
+
+        [QueryKey("direction")]
+        public DirectionValues? DirectionValues { get; set; }
+
+        public override string ToString()
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            Type typeFromHandle = typeof(TreeQuery);
+
+            Postsynaptics?.Select(g => g.ToString()).AppendQuery(typeFromHandle.GetQueryKey("Postsynaptics"), stringBuilder);
+
+            if (RegionId != Guid.Empty)
```

Comment:
Use "this." to reference instance members. Please update all instance member references.

Reply:
DOne